CondCore/BeamSpotPlugins/src/plugin.cc

```diff
@@ -10,4 +10,4 @@
 REGISTER_PLUGIN(BeamSpotObjectsRcd, BeamSpotObjects);
 REGISTER_PLUGIN(SimBeamSpotObjectsRcd, SimBeamSpotObjects);
 REGISTER_PLUGIN(BeamSpotOnlineHLTObjectsRcd, BeamSpotOnlineObjects);
-REGISTER_PLUGIN(BeamSpotOnlineLegacyObjectsRcd, BeamSpotOnlineObjects);
+REGISTER_PLUGIN_NO_SERIAL(BeamSpotOnlineLegacyObjectsRcd, BeamSpotOnlineObjects);
```

Comment:
Why only `BeamSpotOnlineLegacyObjectsRcd` needs to be migrated to `REGISTER_PLUGIN_NO_SERIAL` and not also `BeamSpotOnlineHLTObjectsRcd`?

Reply:
The `REGISTER_PLUGIN` causes the PoolDBESSource objects to be created for the  combination `BeamSpotOnlineLegacyObjectsRcd`/`BeamSpotOnlineObjects` and the new serialization code to be generate the for the class `BeamSpotOnlineObjects` (note, does not care about the record name). The generation of the serialization code for `BeamSpotOnlineObjects` must be done once and only once. The `REGISTER_PLUGIN_NO_SERIAL` only causes the PoolDBESSource objects to be created and does not create the new serialization code.

So to work, one and only one `REGISTER_PLUGIN` for a given C++ class type must exist and if that same C++ class type is associated to additional records in the PoolDBESSource, the other calls need to use `REGISTER_PLUGIN_NO_SERIAL` to avoid duplicating class declarations (which violates C++'s _one definition_ rule).

Reply:
Thanks! understood!
> one and only one REGISTER_PLUGIN for a given C++ class type must exist and if that same C++ class type is associated to additional records in the PoolDBESSource, the other calls need to use REGISTER_PLUGIN_NO_SERIAL

And (thinking about future registrations/implementations of new record) this is enforced by the code not compiling otherwise, correct?

Reply:
Yes, the compile will complain that it has two definitions of the same class (that is actually how I first uncovered this problem :) )